src/libsyntax/util/parser.rs

```diff
@@ -318,6 +317,11 @@ impl ExprPrecedence {
             // Unary, prefix
             ExprPrecedence::Box |
             ExprPrecedence::AddrOf |
+            // Here `let pats = expr` has `let pats =` as a "unary" prefix of `expr`.
+            // However, this is not exactly right. When `let _ = a` is the LHS of a binop we
+            // need parens sometimes. E.g. we can print `(let _ = a) && b` as `let _ = a && b`
+            // but we need to print `(let _ = a) < b` as-is with parens.
+            ExprPrecedence::Let |
```

Comment:
Is this value ever used after introduction of `prec_let_scrutinee_needs_par()`?
If this function returns the incorrect value (`PREC_PREFIX`) it may be accidentally used elsewhere.

Can `prec_let_scrutinee_needs_par()` there be replaced with `ExprPrecedence::Let => AssocOp::LAnd.precedence() as i8` here?

Reply:
Consider:
```rust
a && let _ = c
```
This parses without parens on the RHS.

If we have `ExprPrecedence::Let => AssocOp...` then we get:

```rust
prec =
  rhs_prec_of_op(&&) ->
  prec_of_op(&&) + 1 ->
  6 + 1 =>
  7;

prec_let =
  prec_of_expr(let _ = c) ->
  prec_of_op(&&) ->
  6;

needs_par =
  prec_let < prec ->
  6 < 7 ->
  true;
```

and so we would pretty print as:

```rust
a && (let _ = c)
```

If we use `PREC_PREFIX` here instead we get `needs_par = false`.